streams/shared/src/main/scala/zio/stream/Sink.scala

```diff
@@ -176,120 +176,116 @@ object Sink {
     z: S
   )(costFn: A => Long, max: Long, decompose: A => Chunk[A])(
     f: (S, A) => S
-  ): Sink[Nothing, A, A, S] =
+  ): Sink[Nothing, A, S] =
     ZSink.foldWeightedDecompose(z)(costFn, max, decompose)(f)
 
   /**
    * see [[ZSink.fromEffect]]
    */
-  final def fromEffect[E, B](b: => IO[E, B]): Sink[E, Nothing, Any, B] =
+  final def fromEffect[E, B](b: => IO[E, B]): Sink[E, Any, B] =
```

Comment:
I'm confused. `succeed` gets a type parameter `A`, but this doesn't? How does this compose?

Reply:
Ah good catch. Every sink that has  `A = Any` should be audited. Most of them would need to be polymorphic in `A`. Will fix that.

Reply:
Good catch! 👍